### PR TITLE
fixed zone api issues related to langCode

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java [hunk 1]
```diff
@@ -682,7 +685,7 @@ private UsersDto getUsersFromResponse(ResponseEntity<String> response) {
 				} else
 					dto.setUserName(null);
 				if (null != z.getZoneCode()) {
-					ZoneNameResponseDto zn = zoneservice.getZone(z.getZoneCode(), LanguageUtils.getLanguage());
+					ZoneNameResponseDto zn = zoneservice.getZone(z.getZoneCode(), languageUtils.getDefaultLanguage());
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java:688]: take language code from DTO

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java [hunk 2]
```diff
@@ -672,7 +675,7 @@ public IdResponseDto decommissionMachine(String machineId) {
 		List<String> zoneIds;
 
 		// get user zone and child zones list
-		List<Zone> userZones = zoneUtils.getSubZones(LanguageUtils.getLanguage());
+		List<Zone> userZones = zoneUtils.getSubZones(languageUtils.getDefaultLanguage());
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java:678]: take language code from DTO

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java [hunk 3]
```diff
@@ -488,7 +491,7 @@ private void setMachineTypeNames(List<MachineSearchDto> list) {
 	 */
 	private void setMapStatus(List<MachineSearchDto> list) {
 		List<RegistrationCenter> registrationCenterList = machineUtil.getAllRegistrationCenters();
-		String langCode=LanguageUtils.getLanguage();
+		String langCode=languageUtils.getDefaultLanguage();
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java:494]: take language code from DTO

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java [hunk 4]
```diff
@@ -734,7 +737,7 @@ private String getUserName(String userId) {
 		for (UserCenterMappingExtnDto userCenterMappingExtnDto : userCenterMappingExtnDtos) {
 			UserDetails ud=userDetailsRepository.findUserDetailsById(userCenterMappingExtnDto.getUserId());
 				if(ud!=null) {
-					RegistrationCenter regC=registrationCenterRepository.findByIdAndLangCode(ud.getRegCenterId(),LanguageUtils.getLanguage());
+					RegistrationCenter regC=registrationCenterRepository.findByIdAndLangCode(ud.getRegCenterId(),languageUtils.getDefaultLanguage());
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java:740]: take language code from DTO

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/ZoneUserServiceImpl.java [hunk 5]
```diff
@@ -379,7 +379,7 @@ public StatusResponseDto updateZoneUserMapping(String userId, boolean isActive)
 				} else
 					dto.setUserName(null);
 				if (null != z.getZoneCode()) {
-					ZoneNameResponseDto zn = zoneservice.getZone(z.getZoneCode(),LanguageUtils.getLanguage());
+					ZoneNameResponseDto zn = zoneservice.getZone(z.getZoneCode(),languageUtils.getDefaultLanguage());
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/ZoneUserServiceImpl.java:382]: take language code from DTO

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java [hunk 6]
```diff
@@ -250,8 +253,8 @@ private void searchZones(List<Zone> zones, List<Zone> zoneIds, ZoneUser zu) {
 			List<Zone> langSpecificZones = null;
 			ZoneUser zu=zoneUserRepository.findZoneByUserIdActiveAndNonDeleted(((AuthUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getUserId());
 
-		if (langCode.equals("all")) {
-			String lang=LanguageUtils.getLanguage();
+		if (langCode.equals("all") || langCode==null) {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java:256]: this leads to NPE, put null check before "all" check

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java [hunk 7]
```diff
@@ -273,8 +276,8 @@ private void searchZones(List<Zone> zones, List<Zone> zoneIds, ZoneUser zu) {
 		List<Zone> zones = getZones();
 		List<Zone> langSpecificZones = null;
 		ZoneUser zu=zoneUserRepository.findZoneByUserIdActiveAndNonDeleted(((AuthUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getUserId());
-		if (langCode.equals("all")) {
-			String lang=LanguageUtils.getLanguage();
+		if (langCode.equals("all") || langCode==null) {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java:279]: this leads to NPE, put null check before "all" check

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java [hunk 8]
```diff
@@ -295,8 +298,8 @@ private void searchZones(List<Zone> zones, List<Zone> zoneIds, ZoneUser zu) {
 		List<Zone> zones = getZones();
 		List<Zone> langSpecificZones = null;
 	//	ZoneUser zu=zoneUserRepository.findZoneByUserIdActiveAndNonDeleted(((AuthUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getUserId());
-		if (langCode.equals("all")) {
-			String lang=LanguageUtils.getLanguage();
+		if (langCode.equals("all") || langCode==null) {
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java:301]: this leads to NPE, put null check before "all" check

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java (hunk 1 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java:688]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java (hunk 2 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java:681]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java (hunk 3 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/MachineServiceImpl.java]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java (hunk 4 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/UserDetailsServiceImpl.java:740]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/ZoneUserServiceImpl.java (hunk 5 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/ZoneUserServiceImpl.java:381]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java (hunk 6 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java (hunk 7 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java]: taken care

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java (hunk 8 above)
Comment (dhanendra06) [reply admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/utils/ZoneUtils.java]: taken care

**Review by ase-101 — APPROVED** [review]

**Pull request merged**